Zend/zend_object_handlers.c

```diff
@@ -1117,8 +1117,10 @@ ZEND_API zval *zend_std_get_property_ptr_ptr(zend_object *zobj, zend_string *nam
 							ZSTR_VAL(name));
 						retval = &EG(error_zval);
 					} else {
-						ZVAL_NULL(retval);
 						zend_error(E_WARNING, "Undefined property: %s::$%s", ZSTR_VAL(zobj->ce->name), ZSTR_VAL(name));
+						/* We set the retval to null AFTER the warning so that an error handler cannot mess
+						 * with the property value... */
+						ZVAL_NULL(retval);
```

Comment:
This can leak.

```php
class Foo {}
class C {
    public $a;
    function errorHandler($errno, $errstr) {
        $this->a = new Foo();
    }
}
$c = new C;
set_error_handler([$c,'errorHandler']);
unset($c->a);
(++$c->a);
var_dump($c->a);
```

Just wrapping the `ZVAL_NULL` in a `if (Z_TYPE_P(retval) == IS_UNDEF) {` should do. This will change the behavior to throw a "TypeError: Cannot increment Foo" exception, which I think is what I would expect.

Reply:
I guess if we want that, then for consistency the `zend_hash_update` for the dynamic case should become `zend_hash_add` instead too?

Reply:
@nielsdos That probably makes sense. The leak doesn't happen in that case because the hash table takes care of releasing the old value.

Reply:
OK ty, I'll make a PR soon.

Reply:
@Girgias I think this null should be removed for consistency with https://github.com/php/php-src/commit/a3a3964497922bcd74282685218d90acea69c296.